android/src/main/java/com/appfolio/react/imageutils/ImageResizer.java

```diff
@@ -219,6 +229,39 @@ private static Bitmap loadBitmapFromBase64(@NonNull final Uri imageUri,
         return sourceImage;
     }
 
+    /**
+     * Load a Bitmap from an http or https URL.
+     */
+    @Nullable
+    public static Bitmap loadBitmapFromURL(@NonNull final Uri imageUri,
+                                           @NonNull final BitmapFactory.Options options) throws IOException {
+        Bitmap sourceImage = null;
+        URL url = null;
+
+        try {
+            url = new URL(imageUri.toString());
+        } catch (MalformedURLException e) {
+            throw new MalformedURLException("Unable to load image from a malformed URL: " + e.getMessage());
+        }
+
+        OkHttpClient client = new OkHttpClient();
+        Request request = new Request.Builder()
+            .url(url)
+            .build();
+
+        try (Response response = client.newCall(request).execute()) {
+            if (!response.isSuccessful()) throw new IOException("Unexpected code " + response.code());
+      
+            InputStream input = response.body().byteStream();
+            sourceImage = BitmapFactory.decodeStream(input);
+            input.close();
+        } catch (IOException e) {
+            throw new IOException("Unable to download image: " + e.getMessage());
```

Comment:
What about passing the [original `IOException` as the cause](https://docs.oracle.com/javase/7/docs/api/java/io/IOException.html#IOException(java.lang.String,%20java.lang.Throwable))?

For example `throw new IOException("Unable to download image", e)`.

Reply:
Good idea 👍